src/App.tsx

```diff
@@ -1,15 +1,39 @@
-import React from 'react';
+import { debounce } from 'lodash';
+import React, { useCallback, useMemo, useState } from 'react';
+
 import './App.scss';
+import { Autocomplete } from './components/Autocomplete';
+
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [value, setValue] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [focus, setFocus] = useState(false);
+  const [selectedUser, setSelectedUser] = useState<Person | null>(null);
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, 300), []);
+
+  const handleInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setValue(event.target.value);
+    applyQuery(event.target.value);
+    setSelectedUser(null);
+  };
+
+  const filteredUser = useMemo(() => {
+    return peopleFromServer.filter(el =>
+      el.name.toLowerCase().includes(appliedQuery.toLowerCase()),
+    );
+  }, [appliedQuery]);
```

Comment:
The filtering logic should not be triggered if the `appliedQuery` contains only spaces. Consider adding a check to ensure that the query is not just whitespace before applying the filter.